test cases/common/22 target arg/meson.build

```diff
@@ -1,4 +1,14 @@
-project('local arg test', 'cpp', 'c')
+project('local arg test', 'c')
+
+if add_languages('cpp', required: false)
+  cpp = meson.get_compiler('cpp')
+else
+  error('MESON_SKIP_TEST No C++ compiler available.')
+endif
+
+if host_machine.system() == 'windows' and fcid == 'pgi'
+  error('MESON_SKIP_TEST: PGI on Windows does not have a C++ compiler')
+endif
```

Comment:
ditto

Reply:
For this test, actually the main program is c++ so I think common/22 should indeed be skipped without a C++ compiler available.